### PR TITLE
test for getBlockHash #2534

### PR DESCRIPTION
**Pull request by millken**
#2534

### CODE REVIEW

**Comment by millken** [comment]
*action.Receipt log is always empty?

File: blockchain/integrity/integrity_test.go [hunk 1]
```diff
@@ -558,6 +559,235 @@ func TestCreateBlockchain(t *testing.T) {
...
+	var blockNumber []byte
+	blockNumber = abi.U256(big.NewInt(0))
+	data = append(funcSig[:4], blockNumber...)
+
+	// Add block 2
+	acHash2, err := addOneBlock(contract, 2, zero, gasLimit, gasPrice, data)
+	if err != nil {
+		return err
+	}
+	bcHash2, err := dao.GetBlockHash(2)
```
Comment (dustinxie) [inline blockchain/integrity/integrity_test.go:700]: no need to get bcHash2, and verify it == bc.TipHash()

File: blockchain/integrity/integrity_test.go [hunk 2]
```diff
@@ -558,6 +559,235 @@ func TestCreateBlockchain(t *testing.T) {
...
+		return fmt.Errorf("bcHash2(%x) != bc.TipHash(%x)", bcHash2, bc.TipHash())
+	}
+	fmt.Printf("acHash2 = %x bcHash2 = %x \n", acHash2, bcHash2)
+
+	// Add block 3
+	acHash3, err := addOneBlock(contract, 3, zero, gasLimit, gasPrice, data)
+	if err != nil {
+		return err
+	}
+	bcHash3, err := dao.GetBlockHash(3)
```
Comment (dustinxie) [inline blockchain/integrity/integrity_test.go:714]: same comment

File: blockchain/integrity/integrity_test.go [hunk 3]
```diff
@@ -558,6 +559,235 @@ func TestCreateBlockchain(t *testing.T) {
...
+		return fmt.Errorf("bcHash3(%x) != bc.TipHash(%x)", bcHash3, bc.TipHash())
+	}
+	fmt.Printf("acHash3 = %x bcHash3 = %x \n", acHash3, bcHash3)
+
+	// Add block 4
+	acHash4, err := addOneBlock(contract, 4, zero, gasLimit, gasPrice, data)
+	if err != nil {
+		return err
+	}
+	bcHash4, err := dao.GetBlockHash(4)
```
Comment (dustinxie) [inline blockchain/integrity/integrity_test.go:728]: same comment

File: blockchain/integrity/integrity_test.go [hunk 4]
```diff
@@ -558,6 +559,235 @@ func TestCreateBlockchain(t *testing.T) {
...
+	}
+	if bc.TipHash() != bcHash4 {
+		return fmt.Errorf("bcHash4(%x) != bc.TipHash(%x)", bcHash4, bc.TipHash())
+	}
+	fmt.Printf("acHash4 = %x bcHash4 = %x \n", acHash4, bcHash4)
+
+	// verify getBlockHash
+	//getBlockHash(0)
+	blockNumber = abi.U256(big.NewInt(0))
+	data = append(funcSig[:4], blockNumber...)
```
Comment (dustinxie) [inline blockchain/integrity/integrity_test.go:740]: block, data same as above, can reuse

File: blockchain/integrity/integrity_test.go [hunk 5]
```diff
@@ -558,6 +559,235 @@ func TestCreateBlockchain(t *testing.T) {
...
+
+	acHash5, err := addOneBlock(contract, 5, zero, gasLimit, gasPrice, data)
+	if err != nil {
+		return err
+	}
+	r, err := dao.GetReceiptByActionHash(acHash5, 5)
+	if err != nil {
+		return err
+	}
+	getBlockHash0 := r.Logs()[0].Topics[0]
```
Comment (dustinxie) [inline blockchain/integrity/integrity_test.go:750]: i think can make it a little clean like below:
```go
acHash2 = addOneBlock(, , , , getBlockHash(0)) => equal to block 1
acHash3 = addOneBlock(, , , , getBlockHash(0)) => equal to block 2
acHash4 = addOneBlock(, , , , getBlockHash(1))  => equal to block 2
acHash5 = addOneBlock(, , , , getBlockHash(3))  => equal to block 1

[]tests = []struct {
   acHash
   commitHeight
   targetHeight
}
for range tests {
   r = dao.GetReceiptByActionHash(acHash, commitHeight)
   hash = dao.GetBlockHash(targetHeight)
   r.Logs()[0].Topics[0] should == hash
}
```

**Review by dustinxie — APPROVED** [review]

**Pull request merged**